krm-functions/lib/kubeobject/kubeobject.go

```diff
@@ -0,0 +1,60 @@
+/*
+Copyright 2023 The Nephio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package parser
+
+import (
+	"github.com/GoogleContainerTools/kpt-functions-sdk/go/fn"
+	"sigs.k8s.io/yaml"
+)
+
+type KubeObject[T1 any] struct {
```

Comment:
Should we change the name to KubeObjectExt ( meaning extension). Naming is tough:). Just wanted to distinguish from the base.

Reply:
done